### ISSUE TITLE
Schema name is always null when using @DataSet(value = "empty.yml")

### ISSUE THREAD

**Issue by IgorMaksymov** [database-rider/database-rider#97]
Hello!
I'm trying to clean my tables, without truncating them, with empty dataset and i'm receiving exception:

> NoSuchTableException: The table 'METADATA' does not exist in schema 'null'

The problem located here:
`org.dbunit.database.search.AbstractMetaDataBasedSearchCallback#getNodes(int, java.lang.Object)`

line 146 always contains null
`String schema = this.connection.getSchema();`

the solution is:
`this.connection.getConnection().getSchema()`

as i noticed, this is because of this constructor 
`org.dbunit.database.DatabaseConnection#DatabaseConnection( java.sql.Connection )`

Is there any workaround of this problem ?

I'm using

> rider-spring 1.3.0

**Comment by rmpestano** [comment]
Hi, this code is owned by dbunit project, don't think we can do anything on dbrider side.

You can get in touch with DBUnit team and describe the issue on their mailing list: http://dbunit.sourceforge.net/mail-lists.html

**Comment by IgorMaksymov** [comment]
@rmpestano 
I'm using spring-test-dbunit for years and this happened for the first time.
Also, I've switched back to spring-dbunit.
Also, i've noticed that with database rider comes dbunit 2.5.3, maybe problem hidden there?
Also i may provide my configuration and my project dependencies, maybe there is some conflict.
Strange that Expected dataset works fine, but initial fails

**Comment by rmpestano** [comment]
I don't know if it is the same issue, but we got some problems with schemas, [see here](https://github.com/database-rider/database-rider/issues/49#issuecomment-339776034).

I couldn't investigate further but it can be related.

**Comment by IgorMaksymov** [comment]
@rmpestano meybe its because of Datasource interface implementation or DB?
In my project i use HikariDataSource(comes default with Spring boot) and Postgres

**Comment by human453453467** [comment]
@IgorMaksymov @rmpestano 
I have the same problem. Using Spring Boot(HikariDataSource) with MySQL.
Did anyone solve it?

**Comment by human453453467** [comment]
The problem turned out to be banal for me: wrong name of table. But exception
`org.dbunit.dataset.NoSuchTableException: The table 'WRONG_NAME_TABLE' does not exist in schema 'null'` confuse because of `schema 'null'` but it is normal.

**Comment by rmpestano** [comment]
Hi @ericjturley, do you get the same error with dbunit 2.5.3? (AFAIK we don't support 2.6.0 #111 )

Can you share a simple reproducer?

Thanks for the feedback

**Comment by overstart** [comment]
Hi, i have encounter the same problem. It's the bug of mysql-connector-java, i fixed it by change the version of mysql-connector-java from 8.0.15 to 8.0.17.
https://dev.mysql.com/doc/relnotes/connector-j/8.0/en/news-8-0-17.html.

Hope this is helpful to you

**Comment by metaphor** [comment]
have you guys tried to set `dbunit.yml` with properties,

```
properties:
  caseSensitiveTableNames: true
```

or
```
caseInsensitiveStrategy: !!com.github.database.rider.core.api.configuration.Orthography 'LOWERCASE'
properties:
  caseSensitiveTableNames: false
```

I was using postgresql and with lowercase table name in my dataset yaml, which complained Table Not Found. Because it tried to find UPPERCASE table name.

**Comment by vaa25** [comment]
Guys, seems problem is here https://github.com/database-rider/database-rider/blob/master/rider-core/src/main/java/com/github/database/rider/core/connection/RiderDataSource.java#L130

You don't support MySqlMetadataHandler and NetezzaMetadataHandler from DbUnit. Please fix it asap and release as hot fix version.

**Comment by rmpestano** [comment]
Hi @vaa25, do you have a test reproducer for these databases so we can try to fix the issue?

Also, it would be great if you could contribute the fix as well!

**Comment by dimasmith** [comment]
Hi there. I can confirm the issue. I found some workarounds for that and now trying to pinpoint the cause. I extracted a minimal project to confirm the issue: https://github.com/dimasmith/rider-h2-issue-demo

What's not working for me:
- only xml-based datasets are affected;
- everything works when table names are in uppercase;
- yaml works regardless of table name case;
- `null` schema is not an issue per se;
Note that I only checked for H2.

Workaround for null schema is `dbuint.yml`
```yaml
properties:
  schema: PUBLIC
```

**Comment by rmpestano** [comment]
Hi @dimasmith, thanks for the time digging into this issue and providing a sample project!

The main problem here is that H2 creates tables in uppercase, to create tables with case sensitive names you need to disable uppercase table names on H2, you can do that in application.properties:

```
spring.datasource.url=jdbc:h2:mem:testdb;DATABASE_TO_UPPER=FALSE
```

also you'll need to enable caseSensitiveTableNames in dbunit.yml (or via `@DBUnit` annotation):

```
properties:
  schema: PUBLIC
  caseSensitiveTableNames: true

```
 
This way your first 2 test will pass and the third one will fail because the dataset is in uppercase:

<img width="329" alt="Screenshot 2020-04-14 at 20 41 17" src="https://user-images.githubusercontent.com/1592273/79261663-76b22c00-7e90-11ea-937b-44a3dce67a66.png">

I hope it clarifies your issue

**Comment by rmpestano** [comment]
Also the xml dataset is always case-sensitive, it doesn't take the property `caseSensitiveTableNames` into account

**Comment by dimasmith** [comment]
Hi @rmpestano! Thanks for your response. It's not a severe issue - there are two ways to work around it in my project and you gave me yet another one :) 
From what I can see I can make the failing test pass by setting `useSequenceFiltering = false`.
Looks like it will be great to apply case-insensitive strategy somewhere around the `performSequenceFiltering` in `DataSetExecutorImpl`. But it very well might be that my perception is too narrow and something should be done on a higher level.
Thanks again for the answer and a great tool.

**Comment by rmpestano** [comment]
@dimasmith, can you try with rider `v1.11.2-SNAPSHOT`?

 You'll need to [enable SNAPSHOTS](https://github.com/database-rider/database-rider#snapshots) in order to use it.

**Comment by dimasmith** [comment]
@rmpestano still the same result. I got the rider-core-1.11.2-20200412.055817-2.jar version, might happen it's not the latest snapshot. Updated the project to use the snapshot version. 
I'll be available in few hours, please don't hesitate to ping me if necessary. I'll get back to it ASAP

**Comment by dimasmith** [comment]
@rmpestano - cleaned the maven cache and re-run the build. Everything works now!

**Comment by rmpestano** [comment]
That's great! I'll release 1.11.2 tomorrow.

Thank you very much for your help on this issue!

**Comment by rmpestano** [comment]
This issue was fixed on `v1.11.0` by #192
